pmd-core/src/main/java/net/sourceforge/pmd/properties/PropertyDescriptor.java

```diff
@@ -82,20 +92,15 @@
      *
      * @return A diagnostic message.
      */
-    String errorFor(T value);
+    String errorFor(T value); // TODO Java 1.8 make optional
 
 
     /**
      * Denotes the relative order the property field should occupy if we are using an auto-generated UI to display and
      * edit property values. If the value returned has a non-zero fractional part then this is can be used to place
      * adjacent fields on the same row.
      *
-     * <p>Example:<br> name -&gt; 0.0 description 1.0 minValue -&gt; 2.0 maxValue -&gt; 2.1 </p> ..would have their
-     * fields placed like:<br>
-     *
-     * <code>name: [ ] description: [ ] minimum: [ ] maximum: [ ]</code>
-     *
-     * @return float
+     * @return The relative order compared to other properties of the same rule
      */
     float uiOrder();
```

Comment:
Should we keep this uiOrder? I currently don't give much thought to it, when I define a new property. I guess, it was intended to be used by the eclipse plugin or the designer...
What I don't like about it: it defines something completely unrelated - the presentation layer. The rule will still work, if you set the values of the properties in any arbitrary order...

I've just seen, the compare method is actually wrong if really floats are used:

https://github.com/pmd/pmd/blob/a46331264651423e8a11c748561a5704eca5524b/pmd-core/src/main/java/net/sourceforge/pmd/properties/AbstractProperty.java#L68-L71

We could introduce the convention, that the properties are displayed in the order they are defined.

Reply:
I think that makes sense. So PropertyDescriptor should stop implementing Comparable, and the ordering would instead be defined by the order of Rule.getPropertyDescriptors().

Reply:
definition order? maybe alphabetical order for properties would be best for usability when rendering the documentation

Reply:
@jsotuyod Right I can't see any situation where that wouldn't hold. Related properties usually share a common prefix so they would be close together with alphabetical order.

Reply:
:+1: right, that's the other use case - documentation. Alphabetically sounds fine.